### PR TITLE
story/vspc 3

### PR DESCRIPTION
**Pull request by Neehaarika**
[VSPC-3] User should be able to change the background image of a space.

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java [hunk 1]
```diff
@@ -47,4 +48,19 @@ public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @Reques
 
 		return "redirect:/staff/space/list";
 	}
+	
+	@RequestMapping(value = "/staff/space/update/{id}", method = RequestMethod.POST)
+	public String updateSpace(@PathVariable("id") String id, Model model, @RequestParam("file") MultipartFile file, Principal principal) throws IOException {
+
+		byte[] bgImage = null;
+		String filename = null;
+		if (file  != null) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: what if file is null?

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java [hunk 2]
```diff
@@ -47,4 +48,19 @@ public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @Reques
...
+			bgImage = file.getBytes();
+			filename = file.getOriginalFilename();
+		}
+		ISpace currentSpace = spaceManager.getFullyLoadedSpace(id);
+        spaceManager.storeSpace(currentSpace, bgImage, filename);
+
+		return "redirect:/staff/space/{id}";
+	}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: format/indentation

**Comment by jdamerow** [comment]
also resolve conflicts

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java (hunk 1 above)
Comment (Neehaarika) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: File cant be null, as we alert the user to select the file before hitting change Image button.

**Comment by Neehaarika** [comment]
handling if file is null.

**Comment by Neehaarika** [comment]
Fixed Formatting

**Comment by jdamerow** [comment]
ok to test

**Comment by Neehaarika** [comment]
file empty and indentation changes

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java [hunk 3]
```diff
@@ -20,31 +25,55 @@
...
+            throws IOException {
 
-	@RequestMapping(value = "/staff/space/add", method = RequestMethod.POST)
-	public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @RequestParam("file") MultipartFile file, Principal principal) throws IOException {
+        byte[] bgImage = null;
+        String filename = null;
+        if (file.isEmpty() || file.equals(null)) {
+            attributes.addAttribute("alertType", "warning");
+            attributes.addAttribute("showAlert", "true");
+            attributes.addAttribute("message", "File Not Found");
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: should be "Please select a background image."

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java [hunk 4]
```diff
@@ -20,31 +25,55 @@
...
+            bgImage = file.getBytes();
+            filename = file.getOriginalFilename();
+        }
+        ISpace currentSpace = spaceManager.getFullyLoadedSpace(id);
+        spaceManager.storeSpace(currentSpace, bgImage, filename);
 
-		return "redirect:/staff/space/list";
-	}
+        return new RedirectView(request.getContextPath() + "/staff/space/{id}");
+    }
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: is there a reason you're not returning a string?

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 5]
```diff
@@ -27,19 +29,25 @@ $( document ).ready(function() {
 	    $("#space").append(link);
 	}
 	</c:forEach>
+
+/* 	<c:if test="${param.showAlert eq true}">
+	alert("hi");
+	   Error: ${param.message}
+	</c:if>  */
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp]: remove

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java (hunk 4 above)
Comment (Neehaarika) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: I thought the RedirectAttributes are being leaked to another controller so used RedirectView but understood that's not the case. Changed it back

**Comment by Neehaarika** [comment]
Comments have been addressed.

**Comment by jdamerow** [comment]
ok to test

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 6]
```diff
@@ -77,21 +84,41 @@ $( document ).ready(function() {
...
  <h1>Space: ${space.name}</h1> 
-
+ 
 <div class="alert alert-light" role="alert">
   Created on <span class="date">${space.creationDate}</span> by ${space.createdBy}.
   <br>
-  Modified on <span class="date">${space.modificationDate}</span> by ${space.modifiedBy}.
+  Modified on <span class="date">${space.modificationDate}</span> by ${space.modifiedBy}. 
+  <c:if test="${param.showAlert eq true}">
+  	<div id=errorMsg class="alert alert-${param.alertType}">
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp]: quotation marks missing

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 7]
```diff
@@ -77,21 +84,41 @@ $( document ).ready(function() {
...
+		<h3><strong>ERROR : ${param.message}</strong></h3>
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp]: let's just make that simple text (not heading size and bold). but the alert dialog should be or an error (red) not warning (yellow). and let's not fade it out (especially not so quickly). Just leave the alert there.

File: vspace/src/main/webapp/WEB-INF/tiles/skeleton_bt4.jsp [hunk 8]
```diff
@@ -73,6 +73,11 @@
         <main role="main" class="col-md-9 ml-sm-auto col-lg-10 px-4">
           <div class="row">
           <div class="col-sm" style="padding-top: 20px;">
+            <c:if test="${param.showAlert eq true}">
+          	  <div id="errorMsg" class="alert alert-${param.alertType}">
+			    ERROR : ${param.message}
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/tiles/skeleton_bt4.jsp]: the message might not be necessarily an error.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java [hunk 9]
```diff
@@ -20,31 +25,54 @@
...
+            Principal principal, RedirectAttributes attributes) throws IOException {
 
-	@RequestMapping(value = "/staff/space/add", method = RequestMethod.POST)
-	public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @RequestParam("file") MultipartFile file, Principal principal) throws IOException {
+        byte[] bgImage = null;
+        String filename = null;
+        if (file.isEmpty() || file.equals(null)) {
+            attributes.addAttribute("alertType", "danger");
+            attributes.addAttribute("showAlert", "true");
+            attributes.addAttribute("message", "Please Select a Background Image");
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java]: period at the end of the sentence is missing

**Pull request merged**